lib/jekyll/sitemap_filters.rb

```diff
@@ -0,0 +1,10 @@
+require 'addressable/uri'
+
+module Jekyll
+  module SitemapFilters
+    def normalize_url(input)
+      Addressable::URI.parse(input).normalize.to_s
+    end
+  end
+end
+Liquid::Template.register_filter(Jekyll::SitemapFilters)
```

Comment:
I would rather only include this filter when rendering the sitemap, but nothing that I do seems to make that work 😒